pkg/ebpf/c/tracee.bpf.c

```diff
@@ -2270,13 +2269,16 @@ static __always_inline u64 should_trace(program_data_t *p)
     return p->task_info->matched_scopes;
 }
 
-static __always_inline int should_submit(u32 event_id, config_entry_t *config)
+static __always_inline u64 should_submit(u32 event_id, program_data_t *p)
 {
-    volatile unsigned int index = event_id / 8;
-    unsigned int offset = event_id % 8;
-    u8 bitmap = config->events_to_submit[index % 128];
+    u64 *event_scopes = bpf_map_lookup_elem(&event_scopes_map, &event_id);
```

Comment:
Here we should keep a prior checking against config->events_to_submit, avoiding a map lookup when not necessary.

Reply:
An eBPF program/tail call being attached means some scope needs it submitted almost 100% of the time. As such checking an event in `events_to_submit` first will be true every time the map lookup will succeed and should probably never return false almost ever (this is why we could have almost all kprobes not use should_submit until now).